src/core_plugins/vega/public/vega_view/vega_base_view.js

```diff
@@ -144,16 +180,109 @@ export class VegaBaseView {
 
   setView(view) {
     this._view = view;
+    if (view) {
+      // Global vega expression handler uses it to call custom functions
+      view._kibanaView = this;
+
+      if (this._parser.tooltips) {
+        // position and padding can be specified with
+        // {config:{kibana:{tooltips: {position: 'top', padding: 15 } }}}
+        const tthandler = new TooltipHandler(this._$container[0], view, this._parser.tooltips);
+
+        // Vega bug workaround - need to destroy tooltip by hand
+        this._addDestroyHandler(() => tthandler.hideTooltip());
+      }
+    }
+  }
+
+  /**
+   * Handle
+   * @param funcName
+   * @param args
+   * @returns {Promise<void>}
+   */
+  async vegaFunctionsHandler(funcName, ...args) {
+
+
+
+
+
+    console.log('Executing ' + funcName, ...args);
+
+
+
+
+
+    try {
+      const handlerFunc = vegaFunctions[funcName];
+      if (!handlerFunc || !this[handlerFunc]) {
+        // in case functions don't match the list above
+        throw new Error(`${funcName}() is not defined for this graph`);
+      }
+      await this[handlerFunc].apply(this, args);
+    } catch (err) {
+      this.onError(err);
+    }
+  }
+
+  /**
+   * @param {object} query Elastic Query DSL snippet, as used in the query DSL editor
+   * @param {string} [index] as defined in Kibana, or default if missing
+   */
+  async addFilterHandler(query, index) {
+    const indexId = await this._findIndex(index);
+    const filter = buildQueryFilter(query, indexId);
+    this._queryfilter.addFilters(filter);
```

Comment:
`addFilters` actually returns a promise, that will fail if the filter was invalid. I think we should use this here to show an error message to the user.

```js
try {
  await this._queryfilter.addFilters(filter);
} catch {
  throw new Error('Cannot add filter, since it was invalid.'); // Or another similar error message
}
```